Core/include/Acts/Geometry/TrackingGeometry.hpp

```diff
@@ -145,6 +145,17 @@ class TrackingGeometry {
   const std::unordered_map<GeometryIdentifier, const Surface*>&
   geoIdSurfaceMap() const;
 
+  /// Visualize a tracking geometry including substructure
```

Comment:
Would it be also an option to not couple the `TrackingGeometry` directly to the visualization interface, but rather do it as for surfaces, layers, etc. in `GeometryView3D`? Or are there reasons to do this directly here? (same for `TrackingVolume`)

Reply:
My (original) idea is that `IVisualization3D` is the interface abstraction, and it abstracts the concrete output format. In my opinion, it's fine to have the interface of the geometry objects couple to that. It also (I think) simplifies where this logic lives.

The idea is to remove `GeometryView3D` eventually.

I plan to follow this same pattern for JSON conversion (which is a bit confusing right now).

Reply:
Hmm so I think from my personal taste I would prefer the other way around, because it keeps classes small and focussed, but this is also fine of course.

Reply:
I'm in any case planning to push this further to allow (eventually) full serialization and visualization of volume-internal navigation policies. I think the current approach with lots of downcasting has shown this to be very complex, which I'd like to solve by doing it the other way round.